File: typescript/difference-of-squares/difference-of-squares.ts
export default class Squares {
  squareOfSum: number = 0;
  sumOfSquares: number = 0;
  difference: number = 0;

  constructor(private num: number) {
    if (num > 0) {
      this.squareOfSum = ((num * (num + 1)) / 2) ** 2;
      this.sumOfSquares = (num * (num + 1) * (2 * num + 1)) / 6;
      this.difference = this.squareOfSum - this.sumOfSquares;
    }
  }
}

File: go/difference-of-squares/difference_of_squares.go
// Package diffsquares provides mathematical functions for the squares of natural numbers.
package diffsquares

// SquareOfSum calculates the square of the sum of the first n natural numbers (starting from 1).
func SquareOfSum(n int) int {
	return sum(n) * sum(n)
}

func sum(n int) int {
	return n * (n + 1) / 2
}

// SumOfSquares calculates the sum of the square of the first n natural number (starting from 1).
func SumOfSquares(n int) int {
	return n * (n + 1) * (2*n + 1) / 6
}

// Difference calculates the difference between SquareOfSum and SumOfSquares
func Difference(n int) int {
	return SquareOfSum(n) - SumOfSquares(n)
}
